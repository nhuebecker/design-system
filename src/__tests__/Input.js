import React from 'react'
import renderer from 'react-test-renderer'
import { Input, theme } from '..'

describe('Input', () => {
  test('renders', () => {
    const json = renderer.create(<Input id="renders" />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with custom spacing', () => {
    const json = renderer.create(<Input p={4} m={4} id="space" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('padding', theme.space[4] + 'px')
    expect(json).toHaveStyleRule('margin', theme.space[4] + 'px')
  })

  test('renders with custom fontSize', () => {
    const json = renderer.create(<Input fontSize={4} id="fontSize" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('font-size', theme.fontSizes[4] + 'px')
  })
})
