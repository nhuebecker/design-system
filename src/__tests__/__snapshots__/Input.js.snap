// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  vertical-align: middle;
  max-width: 32rem;
  min-height: 36px;
  line-height: inherit;
  font-family: inherit;
  background-color: transparent;
  border-radius: 4px;
  border-width: 1px;
  border-style: solid;
  border-color: #dde1e4;
  -webkit-transition: 0.1875s cubic-bezier(0.375,0,0.675,1) box-shadow;
  transition: 0.1875s cubic-bezier(0.375,0,0.675,1) box-shadow;
  font-size: 18px;
  margin: 0px;
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
  width: 100%;
  color: inherit;
  background-color: transparent;
}

.c0::-ms-clear {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:focus {
  outline: none;
  border-color: #2d9ee4;
  box-shadow: 0 0 0 2px #add9f5;
}

.c0[type='select'] {
  background: #fff url("data:image/svg+xml;charset=utf8,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 4 5'%3E%3Cpath fill='%23606e77' d='M2 0L0 2h4zm0 5L0 3h4z'/%3E%3C/svg%3E") no-repeat right .75rem center;
  background-size: .5rem;
}

.c0[type='textarea'] {
  resize: vertical;
}

<input
  className="c0"
  id="renders"
/>
`;

exports[`Input renders with custom fontSize 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  vertical-align: middle;
  max-width: 32rem;
  min-height: 36px;
  line-height: inherit;
  font-family: inherit;
  background-color: transparent;
  border-radius: 4px;
  border-width: 1px;
  border-style: solid;
  border-color: #dde1e4;
  -webkit-transition: 0.1875s cubic-bezier(0.375,0,0.675,1) box-shadow;
  transition: 0.1875s cubic-bezier(0.375,0,0.675,1) box-shadow;
  font-size: 27px;
  margin: 0px;
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
  width: 100%;
  color: inherit;
  background-color: transparent;
}

.c0::-ms-clear {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:focus {
  outline: none;
  border-color: #2d9ee4;
  box-shadow: 0 0 0 2px #add9f5;
}

.c0[type='select'] {
  background: #fff url("data:image/svg+xml;charset=utf8,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 4 5'%3E%3Cpath fill='%23606e77' d='M2 0L0 2h4zm0 5L0 3h4z'/%3E%3C/svg%3E") no-repeat right .75rem center;
  background-size: .5rem;
}

.c0[type='textarea'] {
  resize: vertical;
}

<input
  className="c0"
  id="fontSize"
/>
`;

exports[`Input renders with custom spacing 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  vertical-align: middle;
  max-width: 32rem;
  min-height: 36px;
  line-height: inherit;
  font-family: inherit;
  background-color: transparent;
  border-radius: 4px;
  border-width: 1px;
  border-style: solid;
  border-color: #dde1e4;
  -webkit-transition: 0.1875s cubic-bezier(0.375,0,0.675,1) box-shadow;
  transition: 0.1875s cubic-bezier(0.375,0,0.675,1) box-shadow;
  font-size: 18px;
  margin: 24px;
  padding: 24px;
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
  width: 100%;
  color: inherit;
  background-color: transparent;
}

.c0::-ms-clear {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:focus {
  outline: none;
  border-color: #2d9ee4;
  box-shadow: 0 0 0 2px #add9f5;
}

.c0[type='select'] {
  background: #fff url("data:image/svg+xml;charset=utf8,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 4 5'%3E%3Cpath fill='%23606e77' d='M2 0L0 2h4zm0 5L0 3h4z'/%3E%3C/svg%3E") no-repeat right .75rem center;
  background-size: .5rem;
}

.c0[type='textarea'] {
  resize: vertical;
}

<input
  className="c0"
  id="space"
/>
`;
